File: sell-sellergoods-interface/src/main/java/com/synjones/sellergoods/service/BrandService.java
package com.synjones.sellergoods.service;

import com.synjones.pojo.TbBrand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

/**
 * @Author: eric
 * @Description:
 * @Date: 2018/9/7 20:51
 */
public interface BrandService {

    /**
     * 查询全部
     * @return
     */
    public List<TbBrand> findAll();

    /**
     * 分页查询
     */

    public PageResult findPage(int pageNum,int pageSize);

    /**
     * 增加
     */
    public void add(TbBrand brand);

    /**
     * 查询单个
     */
    public TbBrand findOne(long id);

    /**
     * 更新
     */
    public void update(TbBrand brand);

    /**
     * 删除
     */
    public void delete(long[] ids);

    /**
     * 商品map
     */
    public List<Map> selectOptionList();
}

File: sell-manager-web/src/main/webapp/js/service/brandService.js
app.service("brandService",function ($http) {
    //查询全部
    this.findAll= function () {
        return $http.get( '../brand/findAll.do');
    };

    //分页查询
    this.findPage = function (page,size) {
        return $http.get('../brand/findPage.do?page='+page+'&size='+size);
    };

    //保存
    this.add = function (entity) {
        return $http.post('../brand/add.do',entity);
    };

    //查询单个
    this.findOne = function (id) {
        return $http.get('../brand/findOne.do?id='+id);
    };

    //更新
    this.update = function (entity) {
        return $http.post('../brand/update.do',entity);
    };

    //批量删除
    this.delete = function (ids) {
        return  $http.get('../brand/delete.do?ids='+ids);
    }

    //商品列表
    this.selectOptionList=function () {
        return $http.post('../brand/selectOptionList.do');
    }

});